---
title: "cMD Figure 7 rewrite using OmicsMLRepoR Metadata"
author: "Yoon Ji Jung"
date: "2025-04-04"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

7.1 Load R packages
```{r}
suppressPackageStartupMessages({
    library(OmicsMLRepoR)
    library(dplyr)
    library(curatedMetagenomicData)
    library(mia)
    library(scater)
    library(vegan)
    library(stringr)
    library(lefser)
})
```

7.2.1 Retrieve harmonized meta data for cMD using OmicsMLRepoR
```{r, message = FALSE}
cmd <- getMetadata("cMD")
```
7.2.2 Mutate colData and Return Samples 
The association between smoking/smoking history and fecal microbial composition will be examined with smoking status acrooss all relevant samples from curatedMetagenomicData, accessed via returnSamples(). This can be achieved by applying dplyr functions to filter for samples with known smoking status from healthy individuals. After filtering based on these attributes, column containing only NA values are removed from the subset using the where(~ !all(is.na(.x))) statement. Additionally, a new binary variable for smoking status, with levels "Smoker" and "Never Smoker", is created to facilitate downstream analysis.

```{r}
smoke <- cmd %>% 
  tree_filter(disease, "healthy") %>% 
  filter(!is.na(smoker)) %>%
  filter(body_site == "feces") %>%
  select(where(~ !all(is.na(.x))))

smoke <- smoke %>%
  mutate(
    smoker_bin = as.factor(
      case_when(smoker == "Smoker (finding)" ~ "Smoker",
                smoker == "Non-smoker (finding)" ~ "Never Smoker"
      )))

table(smoke$smoker_bin)
```

```{r, message = FALSE}
smoke_tse <- smoke %>% returnSamples("relative_abundance", rownames = "short")
# removing samples with NA values for smoker_bin
smoke_tse <- smoke_tse[,!is.na(smoke_tse$smoker_bin)]
```


7.2.3 Agglomerate By Taxonomic Rank
```{r}
smoke_tse_genus <- agglomerateByRank(smoke_tse, rank = "genus")
```


7.3 Alpha Diversity
Alpha diversity is a measure of the within sample diversity of features (relative abundance proportions here) and seeks to quantify the evenness (i.e. are the amounts of different microbes the same) and richness (i.e. are they are large variety of microbial taxa present). The Shannon index (H’) is a commonly used measure of alpha diversity, it’s estimated here using the addAlpha() function from the mia package.

Using the addAlpha() function from the mia package, alpha diversity between never smokers and smokers can be compared. 
```{r}
# adding shannon_diversity values to colData
smoke_shannon <- smoke_tse_genus %>%
  addAlpha(assay.type = "relative_abundance", index = "shannon_diversity")

# violin plots
smoke_shannon %>% plotColData(x = "smoker_bin", y = "shannon_diversity", colour_by = "smoker_bin", shape_by = "smoker_bin") +
    labs(x = "Smoking Status", y = "Alpha Diversity (H')") +
    guides(colour = guide_legend(title = "Smoking Status"), shape = guide_legend(title = "Alpha Diversity by Smoking Status")) +
    theme(legend.position = "none")

# test if alpha diversity between smokers and non-smokers is significantly different
wilcox.test(shannon_diversity ~ 
              smoker_bin, data = colData(smoke_shannon))
## A p-value < 0.01 and a W value > 0 indicate that the never-smoker group has higher alpha diversity compared to the smoker group. This may serve as basis for further investigation as to whether smoking can lead to gut microbiome dysbiosis.
```
Figure 1. Alpha Diversity - Shannon Index (H')

7.4 Beta Diversity
Beta diversity is a measure of the between sample diversity of features (relative abundance proportions here) and seeks to quantify the magnitude of differences (or similarity) between every given pair of samples. Below it is assessed by Bray–Curtis Principal Coordinates Analysis (PCoA) and Uniform Manifold Approximation and Projection (UMAP).

7.4.1 Bray–Curtis PCoA
To calculate pairwise Bray–Curtis distance for every sample in our study we will use the runMDS() function from the scater package along with the vegdist() function from the vegan package.

To quickly plot the results of beta diversity analysis, the plotReducedDim() function from the scater package is used along with ggplot2 syntax.
```{r}
smoke_tse %>% 
  agglomerateByRanks() %>%
    runMDS(FUN = vegdist, method = "bray", exprs_values = "relative_abundance", altexp = "genus", name = "BrayCurtis") %>%
    plotReducedDim("BrayCurtis", colour_by = "smoker_bin", shape_by = "smoker_bin") +
    labs(x = "PCo 1", y = "PCo 2") +
    guides(colour = guide_legend(title = "Smoking Status"), shape = guide_legend(title = "Smoking Status")) +
    theme(legend.position = c(0.80, 0.25))
```
Figure 2. Beta Diversity – Bray–Curtis PCoA

7.4.2 UMAP
To calculate the UMAP coordinates of every sample in our study we will use the runUMAP() function from the scater package package, as it handles the task in a single line.

To quickly plot the results of beta diversity analysis, the plotReducedDim() function from the scater package is used along with ggplot2 syntax again.
```{r}
smoke_tse %>%
  agglomerateByRanks() %>%
    runUMAP(exprs_values = "relative_abundance", altexp = "genus", name = "UMAP") %>%
    plotReducedDim("UMAP", colour_by = "smoker_bin", shape_by = "smoker_bin") +
    labs(x = "UMAP 1", y = "UMAP 2") +
    guides(colour = guide_legend(title = "Smoking Status"), shape = guide_legend(title = "Smoking Status")) +
    theme(legend.position = c(0.80, 0.55))
```

7.5 Differential Abundance
Next, we can identify which taxa are enriched in either the Smoker or Never Smoker groups. An example approach for differential abundance is the LEfSe analysis,
which can be accomplished using lefser() and lefserPlot() from the lefser package.
```{r}
  lefser(
    relativeAb(smoke_tse_genus),
    kruskal.threshold = 0.05,
    wilcox.threshold = 0.05,
    lda.threshold = 2,
    classCol = "smoker_bin",
    subclassCol = NULL,
    assay = 1L,
    trim.names = FALSE,
    checkAbundances = TRUE
  ) %>%
  lefserPlot()

```

